packages/editor/src/dataviews/store/private-actions.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import type { Action, AnyItem } from '@wordpress/dataviews';
+
+export function registerEntityAction< Item extends AnyItem >(
+	kind: string,
+	name: string,
+	config: Action< Item >
+) {
+	return {
```

Comment:
I know this is the bootstrap PR, but we should add validation, even if it's in a follow up.

Reply:
That's one of the reasons I added the types, for us at least, everything is typed, so there's less need for validation but indeed, validation for third-parties would be great. I wonder if there's a runtime lib that can use typescript types to generate validation automatically.

Reply:
I added a todo list for this to the issue.